### DISCUSSION TITLE
Using both tkinter and ctkinter

### DISCUSSION THREAD

**Discussion by Ryder-07** [TomSchimansky/CustomTkinter#173] [Q&A]
Hello everyone i know it is kind of a silly question but i would like to pose and idea a bout using tkinter and custom tkinter both simultaneously in a single program because it would helpful to use functionality of both of them as custom tkinter is basically a subset so it doesn't have everything but iit is really appreciated that interface and functionalities were improved or maybe i was not able to find if it is already available then please solve this issue for me

**Comment by TomSchimansky** [comment]
It's possible and always was, just try to do it...

**Reply by Ryder-07** [reply]
I have tried and it's not working atleast not the way i am trying

**Reply by felipetesc** [reply]
Perhaps if provide a sample code we can work together to see what's happening.   If we find problems, that's a good thing, because we will be able to make ctk better by fixing the bug

**Comment by felipetesc** [comment]
You can find several samples examplifyng Tkinter and Ctk being used at the same time if you search on the issues. At https://felipetesc.github.io/CtkDocs/#/draggable_frame you can find this sample:

```python
# https://stackoverflow.com/questions/40764168/resize-parts-of-a-python-tkinter-grid
import tkinter as tk

import customtkinter
customtkinter.set_appearance_mode("dark")

def on_clik_print():
	print("bt clicked")


app = customtkinter.CTk()
app.title('Title')
app.geometry("400x300")

panel_divider = tk.PanedWindow(orient="horizontal")
panel_divider.pack(fill="both", expand=True)


left_panel = customtkinter.CTkFrame(master=panel_divider)
left_button = customtkinter.CTkButton(master=left_panel, text="left button", command=on_clik_print)
left_button.place(relx=0.5, rely=0.5, anchor=tk.CENTER)

right_panel = customtkinter.CTkFrame(master=panel_divider)
right_button = customtkinter.CTkButton(master=right_panel, text="right button", command=on_clik_print)
right_button.place(relx=0.5, rely=0.5, anchor=tk.CENTER)

panel_divider.add(left_panel)
panel_divider.add(right_panel)

app.mainloop()

```
After testing this code, as you can see, they work well. The only problem I found so far when using both together, is the fact that they have different styles. To solve this I end up customizing tkinter widgets to match ctk style.

**Reply by felipetesc** [reply]
Can we consider this question answered ?

**Comment by Ryder-07** [comment]
Yes sure i am working on it and if i get into some problem then I'll
connect with you

**Comment by abhaymakes** [comment]
In order to use tkinter and customtkinter widgets both at the same program the main window(`root`) has to be of the `CTk()` and not of `Tk()`. This is because `CTk()` supports widgets from both the program inside it while `Tk()` doesnt. @kunal1209
